### PR TITLE
Add documentation to GitHub pages (Fixes #132)

### LINKED ISSUE #132: Add page for README docs
Add the [README][1] docs to a page into the `docs` folder, to be hosted with Github pages.

More info [here][2]

[1]: https://github.com/mateusjunges/laravel-acl
[2]: https://help.github.com/en/articles/configuring-a-publishing-source-for-github-pages

### PR DESCRIPTION
**Pull request by kimyvgy**
Hello. I see the issue in Hacktoberfest: #132. Let's take a look at the [documentation](https://kimyvgy-forks.github.io/laravel-acl) that I initialized.

### CODE REVIEW

**Comment by mateusjunges** [comment]
I'm in class now. Will review this PR as soon as i get home.

File: docs/installation.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+need to customize the tables, you can publish [the migrations](https://github.com/mateusjunges/laravel-acl/tree/master/src/database/migrations)
+with:
+
+```bash
+php artisan vendor:publish --provider="Junges\ACL\ACLServiceProvider" --tag="acl-migrations"
+```
+and set the `config` for `custom_migrations` to `true`, which is false by default. 
+
+```php
+custom_migrations' => true,
```
Comment (mateusjunges) [inline docs/installation.md]: Missing a single quote here.

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Laravel ACL
+
+This package allows you to manage user permissions and groups in a database.
+
+* [Installation](installation.md)
+    * [Using install command](installation.md#install-using-aclinstall-command)
+    * [Step by step installation](installation.md#step-by-step-installation)
+* [Usage](#usage)
```
Comment (mateusjunges) [inline docs/index.md]: This link is broken.

**Comment by mateusjunges** [comment]
Also, i think it is better to use a custom html and css template to do it. The jekyll theme feels ugly for me xD
You can check [this template][1] built with TailwindCSS. Just download it and modify the content with the documentation of this project.
Thanks in advance for your work, man!

[1]: https://www.tailwindtoolbox.com/templates/minimal-blog

**Comment by kimyvgy** [comment]
> Also, i think it is better to use a custom html and css template to do it. The jekyll theme feels ugly for me xD
> You can check [this template](https://www.tailwindtoolbox.com/templates/minimal-blog) built with TailwindCSS. Just download it and modify the content with the documentation of this project.
> Thanks in advance for your work, man!

I think so. I will update this PR soon

**Comment by mateusjunges** [comment]
Take your time!

**Comment by kimyvgy** [comment]
@mateusjunges How do you think if we use Vuepress instead of the template above?

**Comment by mateusjunges** [comment]
Go ahead :)

**Comment by kimyvgy** [comment]
image

**Comment by mateusjunges** [comment]
@kimyvgy Nice work! Let me now when you are done!

**Comment by kimyvgy** [comment]
@mateusjunges If you want to deploy the documentation with Github pages, I need to built documentation to HTML files then push them to Github in this PR. After that, you can merge it then documentation will display.

**Comment by mateusjunges** [comment]
Got it. If you finished, just build and i'll take a look.

**Comment by kimyvgy** [comment]
@mateusjunges Done. Could you please check it: https://kimyvgy-forks.github.io/laravel-acl/

**Comment by mateusjunges** [comment]
I think it is good. We just need to change [this][1] image. I'll sent one to you tonight, and as soon as you update it, we can merge this PR. Congratulations!

[1]: https://kimyvgy-forks.github.io/laravel-acl/hero.png

**Comment by mateusjunges** [comment]
@kimyvgy Sorry about my late response, but here is [the new image][1] to the page.
If you have time, i have one more request: change the links and buttons color from green to the laravel logo color. This is the RGB code: `rgb(255, 45, 32)`. The hexadecimal color: `#ff2d20`.
Thanks again for your amazing contribution!

[1]: https://drive.google.com/open?id=1-A98gLYHcTvUpR65v3bKO5q4f4UKeXA-

**Comment by kimyvgy** [comment]
@mateusjunges Done. Something went wrong with the SVG image. Therefore, I made a PNG image instead. I also added scripts: `dev:docs` and `build:docs` (with NPM). You can use:
```bash
# In development, access http://localhost:8081
yarn dev:docs

# Build new docs:
yarn build:docs
```

**Comment by mateusjunges** [comment]
@kimyvgy thanks for this contribution and congratulations on your first PR. Welcome to the open source world!

**Comment by mateusjunges** [comment]
@kimyvgy Do i need to build it to publish to github pages? I added the /docs folder on master branch to the github pages config, but it is not showing anything.

**Comment by kimyvgy** [comment]
@mateusjunges Because Github pages do not support the docs in a subfolder of the `/docs`. Therefore, we need to push all of the files in the `/docs/.vuepress/dist` to the `gh-pages` branch and change the Github Pages source option from `/docs` folder to the `gh-pages` branch.

**Comment by mateusjunges** [comment]
Yeah it makes sense. Thanks!

**Pull request merged**